src/Lecture1.hs

```diff
@@ -123,4 +137,17 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+lowerAndGreater :: (Show a, Ord a) => a -> [a] -> [Char]
+lowerAndGreater n list =
+        iter n list 0 0
```

Comment:
Numeric literals in Haskell are polymorphic. That's why you have lots of GHC warnings about defaulting constraints.

To fix this, you can either specify the type of `iter` explicitly or specialize literals here

Reply:
thanks for great instuctions.
Actually I use HLS to specify types and formmating.

Here refctoring `iter`

```haskell
    iter [] lower greater =
      show n
        ++ " is greater than "
        ++ show lower
        ++ " elements and lower than "
        ++ show greater
        ++ " elements"
    iter (h : t) lower greater
      | h < n = iter  t (lower + 1) greater
      | h > n = iter  t lower (greater + 1)
      | otherwise = iter  t lower greater
```

Reply:
Looks great! And yes, HLS can be really helpful 👏🏻